### PR TITLE
fd_send_recv_stubs: replace deprecated C functions

### PR DESCRIPTION
**Pull request by psafont**
This makes the library compile under OCaml 5

Signed-off-by: Pau Ruiz Safont <pau.ruizsafont@cloud.com>

### CODE REVIEW

**Review by lindig — APPROVED** [review]

File: lib/fd_send_recv_stubs.c [hunk 1]
```diff
@@ -36,7 +36,7 @@
 
 void raise_error(int code)
 {
-	static value *exn = NULL;
+	static const value *exn = NULL;
```
Comment (edwintorok) [inline lib/fd_send_recv_stubs.c:39]: We could use `uerror` to raise a Unix_error, unless we rely on the custom exception here for some reason.

File: lib/fd_send_recv_stubs.c [hunk 2]
```diff
@@ -64,7 +64,7 @@ CAMLprim value stub_unix_send_fd(value sock, value buff, value ofs, value len, v
 
   cfd = Int_val(fd);
 
-  cv_flags = convert_flag_list(flags,msg_flag_table);
+  cv_flags = caml_convert_flag_list(flags,msg_flag_table);
```
Comment (edwintorok) [inline lib/fd_send_recv_stubs.c:67]: Can't add a comment on that line specifically but below on line 102 there is a `perror` which might destroy errno, if we want to preserve it then errno should be saved beforehand. However calling `perror` is probably not nice for a library and we'd be better off by just deleting that line.

**Comment by edwintorok** [comment]
The changes look good, however the existing binding could use some cleanup (see comments). Also the bytes/string seem the wrong way around: there is a string function that calles the underlying bytes function with an unsafe conversion, and it'd be better to instead make the string version the safe one and have the bytes  one convert to string (which is safe, since string won't be modified).

File: lib/fd_send_recv_stubs.c (hunk 1 above)
Comment (psafont) [reply lib/fd_send_recv_stubs.c]: Now using `caml_uerror` on both calls that used this function

File: lib/fd_send_recv_stubs.c (hunk 2 above)
Comment (psafont) [reply lib/fd_send_recv_stubs.c:55]: perror removed

**Review by edwintorok — APPROVED** [review]

**Pull request merged**